src/util/charmrun-src/charmrun.C

```diff
@@ -5496,148 +5526,174 @@ static char *find_abs_path(const char *target)
 
 /* simple version of charmrun that avoids the sshd or charmd,   */
 /* it spawn the node program just on local machine using exec. */
-static void start_nodes_local(std::vector<nodetab_process> & process_table)
+struct local_nodestart
 {
-  char ** env = main_envp;
-
-  /* copy environ and expanded to hold NETSTART and CmiNumNodes */
   int envc;
-  for (envc = 0; env[envc]; envc++)
-    ;
-  int extra = 0;
+  char **envp;
+  int n;
+
+  char **dparamp;
+  std::vector<char *> dparamv;
```

Comment:
Could be a std::string as well.

Reply:
Not the way it is used.

Reply:
Yeah, this would require more refactoring.